server/servers/api/src/main/scala/com/prisma/api/schema/Optimizations.scala

```diff
@@ -0,0 +1,126 @@
+package com.prisma.api.schema
+
+import com.prisma.api.connector._
+
+object Optimizations {
+
+  trait Optimizer {
+    def optimize(filter: Filter): Filter
+
+  }
+
+  object FilterOptimizer extends Optimizer {
+
+    override def optimize(filter: Filter): Filter = {
+      val one = LogicalOpt.transform(filter)
+//      val two   = InlineOpt.transform(one)
+      val three = SameRelationFilterOpt.transform(one)
+      three
+    }
+
+    trait Optimization {
+      def transform(filter: Filter): Filter
+    }
+
+    object LogicalOpt extends Optimization {
```

Comment:
I think it would be beneficial to add some explanations of what this filter does. Right now i don't understand the name. From reading it looks to me as if this removes superfluous wrapping of filters. But still i am not sure. This topic seems so involved an explanation is warranted :D

Reply:
That's what it does. Added a comment.